contracts/lazymint/ERC721LazyMint.sol

```diff
@@ -0,0 +1,137 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+/// @author: manifold.xyz
+
+import "@manifoldxyz/creator-core-solidity/contracts/core/IERC721CreatorCore.sol";
+import "@manifoldxyz/libraries-solidity/contracts/access/AdminControl.sol";
+
+import "../libraries/ABDKMath64x64.sol";
+
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import "@openzeppelin/contracts/utils/Strings.sol";
+import "@openzeppelin/contracts/utils/cryptography/MerkleProof.sol";
+
+import "./IERC721LazyMint.sol";
+
+/**
+ * Lazy mint with whitelist ERC721 tokens
+ */
+contract ERC721LazyMint is IERC721LazyMint, ReentrancyGuard {
+  using Strings for uint256;
+  using ABDKMath64x64 for uint;
+
+  // mapping(address => Listing) public listings;
+  // mapping(address => mapping(address => uint256)) private listingTokenEditions;
+  mapping(address => uint) public listingCounts;
+  mapping(address => mapping(uint => Listing)) public listings;
+  mapping(address => mapping(uint => uint)) public mintsPerListing;
+  mapping(address => mapping(uint => mapping(address => uint))) public mintsPerWallet;
+
+  function supportsInterface(bytes4 interfaceId) public view virtual returns (bool) {
+    return interfaceId == type(IERC721LazyMint).interfaceId;
+  }
+
+  modifier creatorAdminRequired(address creatorContractAddress) {
+    // This is shared, not single-creator. So we must ensure that the initializer is an admin of the contract
+    AdminControl creatorCoreContract = AdminControl(creatorContractAddress);
+    require(creatorCoreContract.isAdmin(msg.sender), "Wallet is not an administrator for contract");
+    _;
+  }
+
+  // Initialize
+  function initializeListing(
+    address creatorContractAddress,
+    bytes32 merkleRoot,
+    string calldata uri,
+    uint totalMax,
+    uint walletMax,
+    uint startDate,
+    uint endDate
+  ) external creatorAdminRequired(creatorContractAddress) {
+    // Get the index for the new listing
+    uint newIndex = listingCounts[creatorContractAddress];
+    listingCounts[creatorContractAddress] = newIndex + 1;
+
+    // Create the listing
+    Listing memory newListing = Listing({
+      merkleRoot: merkleRoot,
+      uri: uri,
+      totalMax: totalMax,
+      walletMax: walletMax,
+      startDate: startDate,
+      endDate: endDate
+    });
+    listings[creatorContractAddress][newIndex] = newListing;
```

Comment:
Do we need this in memory here? Wonder if we can save some gas on initializing by just like - 

```
    listings[creatorContractAddress][newIndex] = Listing({
      merkleRoot: merkleRoot,
      uri: uri,
      totalMax: totalMax,
      walletMax: walletMax,
      startDate: startDate,
      endDate: endDate
    });;
```